### DISCUSSION TITLE
Automate required checks on GitHub repos

### DISCUSSION THREAD

**Discussion by sigfrid** [community/community#32189] [Actions]
In my repos, I usually define the same restrictions on the `master` branch as below.

<img width="1427" alt="image" src="https://user-images.githubusercontent.com/3609259/188810260-82286099-d6eb-4855-a89f-7c10cb58bb71.png">

I'm looking for a way to automate it when I create a new repo. 
Is there any infrastructure-as-code (IaC) tool that could help me to achieve it?

**Comment by JackCoplandGranta** [comment]
[You could write something to set those branch protection settings using the rest api](https://docs.github.com/en/rest/branches/branch-protection#update-branch-protection).
